contracts/libraries/ErrorMessage.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+
+pragma solidity >=0.7.0 <0.9.0;
+
+/**
+ * @title Error Message - Contract which uses assembly to revert with a custom error message.
+ * @author Shebin John - @remedcu
+ * @notice The aim is to save gas using assembly to revert with custom error message.
+ */
+abstract contract ErrorMessage {
+    /**
+     * @notice Function which uses assembly to revert with the passed error message.
+     * @param error The error string to revert with.
+     * @dev Currently it is expected that the `error` string is at max 32 bytes of length.
+     */
+    function revertWithError(string memory error) internal pure {
```

Comment:
All revert messages are `bytes5` right? It might make sense to:

```suggestion
    function revertWithError(bytes5 error) internal pure {
```

To additionally avoid allocations

Reply:
Yes, all revert messages are of 5 bytes. Awesome suggestion, this further reduces the overall code size by at least 700+ bytes!